ktor-io/common/src/io/ktor/utils/io/core/Strings.kt

```diff
@@ -89,18 +89,14 @@ public fun Source.readTextExactCharacters(charactersCount: Int, charset: Charset
 }
 
 /**
- * Writes [text] characters in range \[[fromIndex] .. [toIndex]) with the specified [charset]
+ * Writes [text] characters in range [fromIndex] .. [toIndex]) with the specified [charset]
  */
 public fun Sink.writeText(
     text: CharSequence,
     fromIndex: Int = 0,
     toIndex: Int = text.length,
     charset: Charset = Charsets.UTF_8
 ) {
-    if (charset === Charsets.UTF_8) {
-        return writeString(text.toString(), fromIndex, toIndex)
```

Comment:
Could you tell me why whiteStrings corrupts non ASCII? Should we log this to the kotlinx-io instead?